packages/dd-trace/src/ci-visibility/exporters/agentless/writer.js

```diff
@@ -24,28 +52,48 @@ class Writer extends BaseWriter {
       done()
     })
   }
-}
 
-function makeRequest (data, url, cb) {
-  const options = {
-    path: '/api/v2/citestcycle',
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/msgpack',
-      'dd-api-key': process.env.DATADOG_API_KEY || process.env.DD_API_KEY
-    },
-    timeout: 15000
+  _sendPayload (data, _, done) {
+    const options = getRequestOptions(
+      this._url,
+      '/api/v2/citestcycle',
+      { 'Content-Type': 'application/msgpack' }
+    )
+    this._sendPayloadBase(data, options, done)
   }
 
-  options.protocol = url.protocol
-  options.hostname = url.hostname
-  options.port = url.port
+  _sendCoverage (form, done) {
```

Comment:
My original issue with having separate writers was not having multiple classes, but having 2 completely distinct ways to write to the intake. Since this has been implemented exactly this way, multiple writers would make more sense until we can come up with a normalized writer.